src/python/WMCore/MicroService/Unified/Transferor.py

```diff
@@ -143,79 +167,147 @@ def stop(self):
         {
           "wf_A": [record1, record2, ...],
           "wf_B": [....],
         }
         where each record has the following format:
-        {"timestamp":000, "dataset":"/a/b/c", "type": "primary", "trainsferIDs": [1,2,3]}
+        {"timestamp":000, "dataset":"/a/b/c", "dataType": "primary", "transferIDs": [1,2,3]}
+        :param wname: optional workflow name
+        :param dataset: optional dataset name
+        :return: a tranfer ids document
         """
-        doc = {}
+        # meanwhile return transfer ids from internal store
+        doc = self.svc.reqmgrAux.getTransferInfo()
+        if wname and wname in doc:
+            records = doc.get(wname, [])
+            recs = []
+            for rec in records:
+                if dataset:
+                    if dataset == rec.get('dataset', None):
+                        recs.append(rec)
+                else:
+                    recs.append(rec)
+            return {wname: recs}
         return doc
 
-    def updateTransferIDs(self, requestStatus):
-        "Update transfer ids in backend"
-        # TODO/Wait: https://github.com/dmwm/WMCore/issues/9198
-        # doc = self.getTransferIDsDoc()
-
-    def getTransferIDs(self):
-        "Get transfer ids from backend"
-        # TODO/Wait: https://github.com/dmwm/WMCore/issues/9198
-        # meanwhile return transfer ids from internal store 
-        return []
-
-    def getTransferStatuses(self, tids):
-        "get transfer statuses for given transfer IDs from backend"
-        # transfer docs on backend has the following form
-        # https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
-        statuses = {}
-        for tid in tids:
-            # TODO: I need to find request name from transfer ID
-            #status = self.checkSubscription(request)
-            status = 100
-            statuses[tid] = status
-        return statuses
+    def getTransferIds(self, dataset):
+        """
+        Get transfer ids document for given request name and datasets.
+        :param dataset: dataset name
+        :return: a list of transfer ids
+        """
+        # phedex implementation, TODO: implement Rucio logic when it is ready
+        data = self.phedex.subscriptions(dataset=dataset, group=self.group)
+        self.logger.debug("### dataset %s group %s", dataset, self.group)
+        self.logger.debug("### subscription %s", data)
+        tids = []
+        for row in data['phedex']['dataset']:
+            if row['name'] == dataset:
+                for rec in row['subscription']:
+                    # obtain only unfinished transfer ids
+                    if rec['percent_files'] != 100:
+                        tids.append(int(rec['request']))
+        return list(set(tids))
 
-    def requestCampaign(self, req):
-        "Return request campaign"
-        return 'campaign_TODO' # TODO
+    def getStatusRecords(self, reqName):
```

Comment:
Hmm, we don't really persist a status of the transfers. It will be calculated on the fly based on the transfer document. OR, we need to add another key to the trasnfer document schema to keep track of the completion status of those transfers.

Then we still have to define: "completion", but that's in another time.